test/helpers/ProposalTest.sol

```diff
@@ -0,0 +1,336 @@
+
+import {TestableProposeScript} from "test/helpers/TestableProposeScript.sol";
+import {IGovernorAlpha} from "src/interfaces/IGovernorAlpha.sol";
+import {RadworksGovernorTest} from "test/helpers/RadworksGovernorTest.sol";
+
+abstract contract ProposalTest is RadworksGovernorTest {
+  //----------------- State and Setup ----------- //
+
+  IGovernorAlpha governorAlpha = IGovernorAlpha(GOVERNOR_ALPHA);
+  ICompoundTimelock timelock = ICompoundTimelock(payable(TIMELOCK));
+  uint256 initialProposalCount;
+  uint256 upgradeProposalId;
+
+  // As defined in the GovernorAlpha ProposalState Enum
+  uint8 constant PENDING = 0;
+  uint8 constant ACTIVE = 1;
+  uint8 constant DEFEATED = 3;
+  uint8 constant SUCCEEDED = 4;
+  uint8 constant QUEUED = 5;
+  uint8 constant EXECUTED = 7;
+
+  // From GovernorCountingSimple
+  uint8 constant AGAINST = 0;
+  uint8 constant FOR = 1;
+  uint8 constant ABSTAIN = 2;
+
+  function setUp() public virtual override {
+    RadworksGovernorTest.setUp();
+
+    if (_useDeployedGovernorBravo()) {
+      // Use the actual live proposal data when it gets put on chain
+      upgradeProposalId = 17; // assume this is the next proposal
+      // Since the proposal was already submitted, the count before its submissions is one less
+      initialProposalCount = governorAlpha.proposalCount() - 1;
+    } else {
+      initialProposalCount = governorAlpha.proposalCount();
+
+      TestableProposeScript _proposeScript = new TestableProposeScript();
+      // We override the deployer to use an alternate delegate, because in this context,
+      // lonser.eth already has a live proposal
+      _proposeScript.overrideProposerForTests(0x69dceee155C31eA0c8354F90BDD65C12FaF5A00a);
+
+      upgradeProposalId = _proposeScript.run(governorBravo);
+    }
+  }
+
+  //--------------- HELPERS ---------------//
+
+  // a cheat for fuzzing addresses that are payable only
+  // Why is this no longer in standard cheats? JJF
+  // see https://github.com/foundry-rs/foundry/issues/3631
+  function assumePayable(address addr) internal virtual {
+    (bool success,) = payable(addr).call{value: 0}("");
+    vm.assume(success);
+  }
+
+  function _assumeReceiver(address _receiver) internal {
+    assumePayable(_receiver);
+    vm.assume(
+      // We don't want the receiver to be the Timelock, as that would make our
+      // assertions less meaningful -- most of our tests want to confirm that
+      // proposals can cause tokens to be sent *from* the timelock to somewhere
+      // else.
+      _receiver != TIMELOCK
+      // We also can't have the receiver be the zero address because RAD
+      // blocks transfers to the zero address -- see line 396:
+      // https://etherscan.io/address/0x31c8EAcBFFdD875c74b94b077895Bd78CF1E64A3#code
+      && _receiver > address(0)
+    );
+    assumeNoPrecompiles(_receiver);
+  }
+
+  function _randomERC20Token(uint256 _seed) internal pure returns (IERC20 _token) {
```

Comment:
another name that i think is fine as-is, yet here i am trying to optimize. perhaps `_selectERC20Token` expresses more the fact we're pulling these from a list.

Reply:
This was the name used in the PoolTogether code.. keeping for consistency with that.